src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotByRequestIdInterface.java

```diff
@@ -25,15 +23,17 @@ default StepResult prepareAndCreateSnapshot(
       SnapshotDao snapshotDao,
       AuthenticatedUserRequest userReq)
       throws InterruptedException {
+    var workingMap = context.getWorkingMap();
     UUID accessRequestId =
         snapshotReq.getContents().get(0).getRequestIdSpec().getSnapshotRequestId();
     SnapshotAccessRequestResponse accessRequest = snapshotRequestDao.getById(accessRequestId);
 
     UUID sourceSnapshotId = accessRequest.getSourceSnapshotId();
     Snapshot sourceSnapshot = snapshotDao.retrieveSnapshot(sourceSnapshotId);
-    Dataset dataset = sourceSnapshot.getSourceDataset();
+
     AssetSpecification assetSpecification =
-        SnapshotService.getAssetByNameFromDataset(dataset, SnapshotService.ASSET_NAME);
+        workingMap.get(
+            SnapshotWorkingMapKeys.SNAPSHOT_ASSET_SPECIFICATION, AssetSpecification.class);
```

Comment:
Instead of storing it in the map, could we re-generate the asset spec here by calling `SnapshotService.buildAssetFromSnapshotAccessRequest()`?

Is there a need to generate the asset spec in `CreateSnapshotMetadataStep`, or can it be moved to this step instead?

Reply:
We need the asset spec in the CreateSnapshotMetadataStep in order to select which tables are included in the snapshot.
As suggested, I've switched to regenerating the asset spec in both steps by calling that snapshot service method. I couldn't put it in the map because of an infinite recursion error on attempting to put the asset specification into the flight map.